python/lsst/ts/ess/csc/accumulator/air_flow_accumulator.py

```diff
@@ -0,0 +1,178 @@
+    the ``num_bad_samples`` counter. In the unlikely event that we accumulate
+    ``num_samples`` of bad data before that many good samples,
+    ``do_report()`` will be true, but ``get_topic_kwargs()`` will return
+    ``nan`` for all statistical values. The point is to publish *something*,
+    since this is telemetry and it should be output at semi-regular intervals.
+    Note that the accumulated good data will be lost.
+    """
+
+    def __init__(self, num_samples: int) -> None:
+        if num_samples < 2:
+            raise ValueError(f"{num_samples=} must be > 1")
+        self.num_samples = num_samples
+        self.timestamp: list[float] = list()
+        self.speed: list[float] = list()
+        self.direction: list[int] = list()
+        self.num_bad_samples = 0
+
+    @property
+    def do_report(self) -> bool:
+        """Do we have enough data to report good or bad data?"""
+        return max(len(self.speed), self.num_bad_samples) >= self.num_samples
+
+    def add_sample(
+        self,
+        timestamp: float,
+        speed: float,
+        direction: int,
+        isok: bool,
+    ) -> None:
+        """Add a sample.
+
+        Parameters
+        ----------
+        timestamp : `float`
+            Time at which data was taken (TAI unix seconds).
+        speed : `float`
+            Wind speed (m/s).
+        direction : `int`
+            Wind direction (deg).
+        isok : `bool`
+            Is the data valid?
+        """
+        if isok:
+            self.timestamp.append(timestamp)
+            self.speed.append(speed)
+            self.direction.append(direction)
+        else:
+            self.num_bad_samples += 1
+
+    def clear(self) -> None:
+        """Clear the accumulated data.
+
+        Note that ``get_topic_kwargs()`` automatically calls this,
+        so you typically will not have to.
+        """
+        self.timestamp = list()
+        self.speed = list()
+        self.direction = list()
+        self.num_bad_samples = 0
+
+    def get_topic_kwargs(self) -> dict[str, float | list[float] | bool]:
+        """Return data for the electricFieldStrength telemetry topic.
+
+        Returns
+        -------
+        topic_kwargs : `dict` [`str`, `float`]
+            Data for the electricFieldStrength telemetry topic as a keyword,
+            arguments, or an empty dict if there are not enough samples yet.
+            A dict with data will have these keys:
+
+            * strength
+            * strengthStdDev
+            * strengthMax
+        """
+        timestamp = self.timestamp[-1]
+        if len(self.speed) >= self.num_samples:
+            # Return good data
+            speed_arr = np.array(self.speed)
+            speed_median = get_median(data=speed_arr)
+            speed_std = np.std(speed_arr)
+            direction_arr = np.array(self.direction)
+            direction_median = int(get_median(data=direction_arr))
+            direction_std = int(np.std(direction_arr))
```

Comment:
Please delete the `int` casts in the above two lines. I see that the field values in ts_xml "short" but we really need to fix that, and until we do, I think it's best to do an explicit cast when writing the telemetry topic -- to make it crystal clear that we are accommodating surprising XML.

(Sorry if that was me who chose "short" for those values. I'll be happy to fix that!)

Reply:
If you have time, please use quartiles to compute standard deviation for all the accumulators. That is what Robert Lupton ended up recommending. std dev ~= 0.741*(Q75 - Q25) assuming a gaussian distribution. It's more self-consistent than reporting median and real standard deviation. I should have filed a ticket about that, but I think I forgot. You can leave it for another ticket if you prefer, but I think it's not a big stretch to include it here.

To do make this change, consider changing get_median to return median and this estimate of std dev, and rename the function accordingly.

If you do that, then please either update the descriptions in ts_xml or I will be happy (on this ticket or a new one, as you prefer).

if you want to keep reporting real std dev then please get rid of the get_median function and just call np.median (or np.quantile with [0.5]). It is a inefficient to call np.quantile and ask for 3 quantiles and then discard two of them.

Reply:
I updated get_median to now be get_median_and_std_dev and filed DM-37648 to update ts_xml and assigned it to myself.